### ISSUE TITLE
CDN

### ISSUE THREAD

**Issue by NekR** [NekR/offline-plugin#84]
It's common to use CDN when you develop web applications and other that CDN lives on separate sub-domain, like `cdn.example.com`.

There is no problem handling CDN resources with ServiceWorker (except that those resources must have CORS headers). But with AppCache it isn't essentially possible to cache cross-domain resources over HTTPS.

One possible solution is to generate 2 manifests: 1 for main domain (the app) and 2 for the CDN.
Roughly, it can look like this:

```js
new OfflinePlugin({
  cdn: {
    assets: ['*.js']
  }
})
```

**Comment by navgarcha** [comment]
Is the problem that if you utilise `mode: no-cors` that the non local assets are responding with `response.ok === false`? As this is what I'm getting? If this is the case do we have to ensure all assets respond with CORS headers?

**Comment by NekR** [comment]
> Is the problem that if you utilise mode: no-cors that the non local assets are responding with response.ok === false? As this is what I'm getting. If this is the case do we have to ensure all assets respond with CORS headers?

Yes, exactly. With `mode: no-cors` `response.ok` is `false` for cross-origin requests unless those responses provide correct CORS headers.

> do we not ALWAYS get response.ok === false even on successful requests. In this situation the logic to check response.ok will be inaccurate

It's accurate is possible. You don't response status there and don't want to cache 404 or 500 or other responses with error status. Hence, not allowing to cache `opaque` responses. Otherwise users may get broken app for a very long time and you certainly don't want it.

>  Granted I don't really know what the knock on affects are of caching a potential 404 response..

If you cache 404 response, you always server assets with 404 from the cache.

**Comment by navgarcha** [comment]
That makes sense - was thinking out loud slightly there :) 

Perhaps a note in the README or a more informative error message might be useful around the 'no-cors' mode?

**Comment by NekR** [comment]
@navgarcha Yeah, I think it makes sense to add it to README or to the FAQ. Would you mind to do the PR?

**Comment by navgarcha** [comment]
Done: https://github.com/NekR/offline-plugin/pull/158
